api/metrics/src/main/java/io/opentelemetry/api/metrics/BoundDoubleCounter.java

```diff
@@ -5,20 +5,28 @@
 
 package io.opentelemetry.api.metrics;
 
+import io.opentelemetry.context.Context;
 import javax.annotation.concurrent.ThreadSafe;
 
-/** A {@code Bound Instrument} for a {@link DoubleCounter}. */
+/** A counter instrument that records {@code double} values with preassociated attributes. */
 @ThreadSafe
-public interface BoundDoubleCounter extends BoundSynchronousInstrument {
+public interface BoundDoubleCounter {
   /**
-   * Adds the given {@code increment} to the current value. The values cannot be negative.
+   * Reecord a value with prebound attributes.
```

Comment:
Perhaps not really a point for this PR and more for the spec (but I believe feedback loop is faster on language SIG PRs than spec issues :-D) but what's the use case for prebound attributes given we have `Resource`? We haven't written that many metrics in instrumentation yet, but so far I haven't seen any time I would want to use prebound (having the Resource is important of course to handle "static attributes")

Reply:
Just ran performance benchmarks.  Pre-bound synchronous instruments are 2x faster due to lacking 1 of the lock-and-access paths for the `ConcurrentMap`.

It's literally just a performance optimisation (and it's used inside the SDK)

Reply:
Yeah I'm sure it will perform better to prebind attributes vs recording each time when the attributes are known. I'm wondering whether we have any examples of attributes that are known ahead of time, that aren't in the Resource?

Also, while writing a unit test for metrics, I found it a bit unnatural for bound instruments to have a different API then unbound. How about allowing attributes to be passed in even for bound instruments? In most cases maybe they won't be passed in, but API-wise I'm not sure of a reason to forbid it.

Reply:
Nevermind, figured out how to use the bound counters.